Plugins/Detray/include/Acts/Plugins/Detray/DetrayConversionHelper.hpp

```diff
@@ -0,0 +1,117 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2023 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Surfaces/SurfaceBounds.hpp"
+#include "Acts/Utilities/BinningData.hpp"
+
+#include <string>
+#include <tuple>
+
+#include <nlohmann/json.hpp>
+
+#include "detray/io/frontend/definitions.hpp"
+
+namespace Acts::DetrayConversionHelper {
+
+/// @brief Helper function to switch keys from ACTS to detray
+///
+/// DETRAY types @todo change to detray imports when available
+///    annulus2 = 0u,
+///    cuboid3 = 1u,
+///    cylinder2 = 2u,
+///    cylinder3 = 3u,
+///    portal_cylinder2 = 4u,
+///    rectangle2 = 5u,
+///    ring2 = 6u,
+///    trapezoid2 = 7u,
+///    cell_wire = 8u,
+///    straw_wire = 9u,
+///    single1 = 10u,
+///    single2 = 11u,
+///    single3 = 12u,
+///    unknown = 13u
+///
+/// @param sBounds is the surface bounds type
+/// @param portal is the flag for conversion into detray portal format
+///
+/// @return type and value array in detray format
+inline static std::tuple<unsigned int, std::vector<ActsScalar>> maskFromBounds(
+    const Acts::SurfaceBounds& sBounds, bool portal = false) {
```

Comment:
Could all of these use dynamic_casts to the derived classes and access the bound values through the respective enums and the getter method? This makes this a bit less brittle versus accessing arbitrary elements from a vector of arbitrary size without bounds checking.